native/cocos/scene/RenderWindow.cpp

```diff
@@ -143,6 +136,32 @@ void RenderWindow::extractRenderCameras(ccstd::vector<Camera *> &cameras) {
     }
 }
 
+void RenderWindow::onNativeWindowDestroy(uint32_t windowId) {
+    if (_swapchain != nullptr && _swapchain->getWindowId() == windowId) {
+        _swapchain->destroySurface();
+    }
+}
+
+void RenderWindow::onNativeWindowResume(uint32_t windowId) {
+    if (_swapchain == nullptr || _swapchain->getWindowId() != windowId) {
+        return;
+    }
+    auto *windowMgr = BasePlatform::getPlatform()->getInterface<ISystemWindowManager>();
+    auto *hWnd = reinterpret_cast<void *>(windowMgr->getWindow(windowId)->getWindowHandle());
+    _swapchain->createSurface(hWnd);
+    updateFramebuffer();
+}
+
+void RenderWindow::updateFramebuffer() {
+    if (_frameBuffer != nullptr) {
+        _frameBuffer = nullptr;
+    }
```

Comment:
I think it is not needed as IntrusivePtr will handle it when be assigned to a new value.